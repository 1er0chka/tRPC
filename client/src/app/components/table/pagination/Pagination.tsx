import React, { FunctionComponent, useEffect, useState } from "react";
import styles from "./Pagination.module.scss";
import Button from "@/app/components/button/Button";

interface IPaginationProps {
  refreshTable: (a: number) => Promise<void>;
  itemsNumber: number;
}

const Pagination: FunctionComponent<IPaginationProps> = ({
  refreshTable,
  itemsNumber,
}) => {
  const [offset, setOffset] = useState<number>(0);

  const increaseOffset = () => {
    setOffset(offset + 40);
  };

  const decreaseOffset = () => {
    setOffset(offset - 40);
  };

  useEffect(() => {
    refreshTable(offset);
  }, [offset]);

  return (
    <div className={styles.pagination}>
      <Button onClick={decreaseOffset} disabled={offset - 40 < 0} text={"<"} />
      <div>
        {offset} - {offset + 40 > itemsNumber ? itemsNumber : offset + 40}
      </div>
      <Button
        onClick={increaseOffset}
        disabled={offset + 40 > itemsNumber}
        text={">"}
      ></Button>
    </div>
  );
};

export default Pagination;
