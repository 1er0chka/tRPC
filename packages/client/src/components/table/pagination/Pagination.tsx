import React, {FunctionComponent, useEffect, useState} from "react";
import styles from "./Pagination.module.scss";
import Button from "../../button/Button";

interface IPaginationProps {
    refreshTable: (a: number) => Promise<void>;
    itemsNumber: number;
}

const Pagination: FunctionComponent<IPaginationProps> = ({
                                                             refreshTable,
                                                             itemsNumber,
                                                         }) => {
    const [offset, setOffset] = useState<number>(0);

    const increaseOffset = () => {
        setOffset(offset + 40);
    };

    const decreaseOffset = () => {
        setOffset(offset - 40);
    };

    useEffect(() => {
        refreshTable(offset);
    }, [offset]);

    return (
        <div className={styles.pagination} data-testid="pagination">
            <Button onClick={decreaseOffset} disabled={offset - 40 < 0} label={"<"} mode={"primary"} dataTestId={"back-button"}/>
            <div data-testid="coins-table-pagination">
                {offset} - {offset + 40 > itemsNumber ? itemsNumber : offset + 40}
            </div>
            <Button
                onClick={increaseOffset}
                disabled={offset + 40 > itemsNumber}
                label={">"} mode={"primary"}
                dataTestId={"forward-button"}
            ></Button>
        </div>
    );
};

export default Pagination;
