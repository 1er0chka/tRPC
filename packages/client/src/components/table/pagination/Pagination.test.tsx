import {render, screen} from "@testing-library/react";
import Pagination from "./Pagination";

const itemsNumber = 40
const refreshTable = jest.fn()

describe('Pagination Tests', () => {
    test('Snapshot test', () => {
        render(
            <Pagination itemsNumber={itemsNumber}  refreshTable={refreshTable}/>
        )
        const pagination = screen.queryByTestId('pagination')
        expect(pagination).toMatchSnapshot()
    })
    test('Back test', () => {
        render(
            <Pagination itemsNumber={itemsNumber}  refreshTable={refreshTable}/>
        )
        const button = screen.queryByTestId('back-button')
        if (button){
            button.click()
        }
    })
    test('Forward test', () => {
        render(
            <Pagination itemsNumber={itemsNumber}  refreshTable={refreshTable}/>
        )
        const button = screen.queryByTestId('forward-button')
        if (button){
            button.click()
        }
    })
})