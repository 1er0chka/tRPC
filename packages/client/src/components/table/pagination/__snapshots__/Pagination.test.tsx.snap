// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination Tests Snapshot test 1`] = `
<div
  class="pagination"
  data-testid="pagination"
>
  <button
    class="primaryButton"
    data-testid="back-button"
    disabled=""
  >
    &lt;
  </button>
  <div
    data-testid="coins-table-pagination"
  >
    0
     - 
    40
  </div>
  <button
    class="primaryButton"
    data-testid="forward-button"
  >
    &gt;
  </button>
</div>
`;
